plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchExecutor.groovy

```diff
@@ -265,6 +271,28 @@ class AwsBatchExecutor extends Executor implements ExtensionPoint {
     @PackageScope
     ThrottlingExecutor getReaper() { reaper }
 
+    void killTask(String jobId) {
+        // extract job array id
+        if( jobId.contains(':') )
+            jobId = jobId.split(':')[0]
+
+        // prevent duplicate delete requests on the same job
+        if( jobId in deletedJobs )
+            return
+        else
+            deletedJobs.add(jobId)
+
+        // submit terminate request
+        reaper.submit({ killTask0(jobId) })
+    }
+
+    protected void killTask0(String jobId) {
+        final req = new TerminateJobRequest()
+            .withJobId(jobId)
+            .withReason('Job killed by NF')
+        final resp = client.terminateJob(req)
```

Comment:
Note that previously it was using `bypassProxy` to access directly to the client, otherwise a kind of deadlock can arise (there should be a ticket for it)

https://github.com/nextflow-io/nextflow/blob/73015fbdb8624e146866d3202c610c9e319f54ee/plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchTaskHandler.groovy#L304-L304